src/components/RoadTabs.vue

```diff
@@ -149,16 +161,22 @@ export default {
     createRoad: function () {
       if (!this.duplicateRoad) {
         this.$emit('add-road', this.newRoadName);
+        this.addDialog = false;
+        this.newRoadName = '';
       } else if (this.duplicateRoadSource in this.roads) {
-        this.$emit('add-road',
-          this.newRoadName,
-          this.roads[this.duplicateRoadSource].contents.coursesOfStudy.slice(0),
-          this.roads[this.duplicateRoadSource].contents.selectedSubjects.map((semester) => semester.slice(0)),
-          Object.assign({}, this.roads[this.duplicateRoadSource].contents.progressOverrides)
-        );
+        if (this.$store.state.unretrieved.indexOf(this.duplicateRoadSource) >= 0) {
+          this.$emit('retrieve', this.duplicateRoadSource);
+        } else {
+          this.$emit('add-road',
+            this.newRoadName,
+            this.roads[this.duplicateRoadSource].contents.coursesOfStudy.slice(0),
+            this.roads[this.duplicateRoadSource].contents.selectedSubjects.map((semester) => semester.slice(0)),
+            Object.assign({}, this.roads[this.duplicateRoadSource].contents.progressOverrides)
+          );
+          this.addDialog = false;
+          this.newRoadName = '';
+        }
       }
-      this.addDialog = false;
-      this.newRoadName = '';
```

Comment:
why duplicate this at the end of both if/else branches?

Reply:
it's only in 2/3 branches; it doesn't happen if you need to retrieve the road

Reply:
oh oops, you're totally right